src/sdk/conference/quicconnection.js

```diff
@@ -0,0 +1,246 @@
+      case 'stream':
+        this._onStreamEvent(message);
+        break;
+      default:
+        Logger.warning('Unknown notification from MCU.');
+    }
+  }
+
+  async _init() {
+    const receiveStreamReader =
+        this._quicTransport.receiveStreams().getReader();
+    Logger.info('Reader: ' + receiveStreamReader);
+    let receivingDone = false;
+    while (!receivingDone) {
+      const {value: receiveStream, done: readingReceiveStreamsDone} =
+          await receiveStreamReader.read();
+      Logger.info('New stream received');
+      if (readingReceiveStreamsDone) {
+        receivingDone = true;
+        break;
+      }
+      const chunkReader = receiveStream.readable.getReader();
+      const {value: uuid, done: readingChunksDone} = await chunkReader.read();
+      if (readingChunksDone) {
+        Logger.error('Stream closed unexpectedly.');
+        return;
+      }
+      if (uuid.length != 16) {
+        Logger.error('Unexpected length for UUID.');
+        return;
+      }
+      chunkReader.releaseLock();
+      const subscriptionId = this._uint8ArrayToUuid(uuid);
+      this._quicStreams.set(subscriptionId, receiveStream);
+      if (this._subscribePromises.has(subscriptionId)) {
+        const subscription =
+            this._createSubscription(subscriptionId, receiveStream);
+        this._subscribePromises.get(subscriptionId).resolve(subscription);
+      }
+    }
+  }
+
+  _createSubscription(id, receiveStream) {
+    // TODO: Incomplete subscription.
+    const subscription = new Subscription(id, () => {
+      receiveStream.abortReading();
+    });
+    subscription.stream = receiveStream;
+    return subscription;
+  }
+
+  async _authenticate(token) {
+    await this._quicTransport.ready;
+    const quicStream = await this._quicTransport.createSendStream();
+    const writer = quicStream.writable.getWriter();
+    await writer.ready;
+    writer.write(new Uint8Array(16));
+    const encoder = new TextEncoder();
+    writer.write(encoder.encode(token));
+    Logger.info('End of auth.');
+  }
+
+  async createSendStream() {
+    await this._quicTransport.ready;
+    const quicStream = await this._quicTransport.createSendStream();
+    return quicStream;
+  }
+
+  async createSendStream1(sessionId) {
+    Logger.info('Create stream.');
+    await this._quicTransport.ready;
+    // TODO: Potential failure because of publication stream is created faster
+    // than signaling stream(created by the 1st call to initiatePublication).
+    const publicationId = await this._initiatePublication();
+    const quicStream = await this._quicTransport.createSendStream();
+    const writer = quicStream.writable.getWriter();
+    await writer.ready;
+    writer.write(this._uuidToUint8Array(publicationId));
+    writer.releaseLock();
+    this._quicStreams.set(publicationId, quicStream);
+    return quicStream;
+  }
+
+  async publish(stream) {
+    // TODO: Avoid a stream to be published twice. The first 16 bit data send to
```

Comment:
I think this is a workaround. Can we remove that requirement.

Reply:
This is not a workaround. It's by design. Please see https://github.com/jianjunz/owt-server/blob/quic-protocol/doc/proposals/quic-transport-payload-format.md.